a3fe/configuration/engine_config.py

```diff
@@ -326,17 +276,96 @@ def get_somd_config(self, run_dir: str, config_name: str = "somd_config") -> str
             "",
             "### Alchemistry ###",
             f"perturbed residue number = {self.perturbed_residue_number}",
-            f"energy frequency = {self.energy_frequency}"
-        ]
+            f"energy frequency = {self.energy_frequency}",
+            f"ligand charge = {self.ligand_charge}",
+            f"charge difference = {self.charge_difference}",
+            "",
+            "### Restraints ###",
+            f"use boresch restraints = {self.use_boresch_restraints}",
+            f"turn on receptor-ligand restraints mode = {self.turn_on_receptor_ligand_restraints}"
+        ])
+        # 2) Lambda parameters
+        config_lines.extend(["", "### Lambda / Alchemical Settings ###"])
+        
+        if self.lambda_array:
+            lambda_str = f"[{', '.join(str(x) for x in self.lambda_array)}]"
+            config_lines.append(f"lambda_array = {lambda_str}")
+        if self.lambda_val is not None:
+            config_lines.append(f"lambda_val = {self.lambda_val}")
+
+        # 3) Alchemical files path
+        config_lines.extend(["", "### Alchemical Files ###"])
+        if self.morphfile:
+            config_lines.append(f"morphfile = {self.morphfile}")
+        if self.topfile:
+            config_lines.append(f"topfile = {self.topfile}")
+        if self.crdfile:
+            config_lines.append(f"crdfile = {self.crdfile}")
+
+        # 5) Boresch restraints 
+        if self.boresch_restraints_dictionary is not None:
+            config_lines.extend(["", "### Boresch Restraints Dictionary ###"])
+            config_lines.append(f"boresch restraints dictionary = {self.boresch_restraints_dictionary}")
+
         # Add any extra options
         if self.extra_options:
             config_lines.extend(["", "### Extra Options ###"])
             for key, value in self.extra_options.items():
                 config_lines.append(f"{key} = {value}")
 
         # Write the configuration to a file
-        config_path = _os.path.join(run_dir, f"{config_name}.cfg")
+        config_path = _os.path.join(run_dir, config_filename)
         with open(config_path, "w") as f:
             f.write("\n".join(config_lines) + "\n")
+        
+        return config_path
+
+    @classmethod
```

Comment:
Great idea! I think we should never use this in the a3fe code, but it will super helpful for users to convert old somd.cfg files to the pydantic configs.